velodyne_pointcloud/src/lib/rawdata.cc

```diff
@@ -175,6 +177,14 @@ inline float SQR(float val) { return val*val; }
 
       // upper bank lasers are numbered [0..31]
       // NOTE: this is a change from the old velodyne_common implementation
+      if (config_.model == "64E_S3")
+      {
+        if (i % 4 < 1) {       //only return strongest return NOTE: for 64E_S3 dual return mode.
+          continue;
+        }
+      }
+
+
```

Comment:
I have a problem with assuming that the 64E_S3 is in dual-return mode when all other models currently make the assumption that this *specifically* is not the case. Adding support for dual-return mode is in the pipeline (see #198) but is not currently supported by any other model.